packages/backend/src/peripherals/discord/DiscordClient.ts

```diff
@@ -4,30 +4,38 @@ https://discord.com/developers/docs/getting-started#configuring-a-bot
 */
 
 import { HttpClient } from '@l2beat/shared'
+import { RateLimiter } from '@l2beat/shared-pure'
 import { RequestInit } from 'node-fetch'
 import { Counter } from 'prom-client'
 
+import { DiscordConfig } from '../../config/Config'
+
 export const MAX_MESSAGE_LENGTH = 2000
 
 export type Channel = 'PUBLIC' | 'INTERNAL'
 
 export class DiscordClient {
   constructor(
     private readonly httpClient: HttpClient,
-    private readonly discordToken: string,
-    private readonly publicChannelId: string | undefined,
-    private readonly internalChannelId: string,
-  ) {}
+    private readonly config: DiscordConfig,
+  ) {
+    if (config.callsPerMinute) {
+      const rateLimiter = new RateLimiter({
+        callsPerMinute: config.callsPerMinute,
+      })
+      this.sendMessage = rateLimiter.wrap(this.sendMessage.bind(this))
```

Comment:
I think you want to wrap `query`. 
```suggestion
      this.query = rateLimiter.wrap(this.query.bind(this))
```